src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,371 @@
+            None
+            if payload["data"]["object"]["trial_end"] == "null"
+            else timezone.make_aware(
+                datetime.fromtimestamp(payload["data"]["object"]["trial_end"])
+            )
+        )
+        assert subscription.plan == bool(
+            payload["data"]["object"]["items"]["data"][0]["plan"]["active"]
+        )
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_invoice_paid(self, mock_construct_event, client, user):
+        """
+        Test the `invoice.paid` event.
+        Verify that the `hosted_invoice_url` and `invoice_pdf` fields
+        of the subscription are correctly updated.
+        """
+        payload = {
+            "type": "invoice.paid",
+            "data": {
+                "object": {
+                    "customer": "cus_REbNQXKKFCRF2c",
+                    "hosted_invoice_url": "https://stripe.com/invoice/test123",
+                    "invoice_pdf": "https://stripe.com/invoice/test123.pdf",
+                }
+            },
+        }
+
+        CustomerDetail.objects.create(
+            user=user, stripe_customer_id=payload["data"]["object"]["customer"]
+        )
+
+        Subscription.objects.create(
+            user=user, status="active", hosted_invoice_url="", invoice_pdf=""
+        )
+
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        mock_construct_event.return_value = payload
+
+        response = client.post(
+            reverse_lazy("v1:payment:stripe-webhook"),
+            data=json.dumps(payload),
+            content_type="application/json",
+            HTTP_STRIPE_SIGNATURE=sig_header,
+        )
+
+        assert response.status_code == 200
+
+        subscription = Subscription.objects.get(user=user)
+        assert (
+            subscription.hosted_invoice_url
+            == payload["data"]["object"]["hosted_invoice_url"]
+        )
+        assert subscription.invoice_pdf == payload["data"]["object"]["invoice_pdf"]
+
+    @mock.patch("stripe.Webhook.construct_event")
+    def test_customer_subscription_deleted(self, mock_construct_event, client, user):
+        """
+        Test the `customer.subscription.deleted` event.
+        Verify that the subscription status is correctly updated to `Canceled`.
+        """
+        payload = {
+            "type": "customer.subscription.deleted",
+            "data": {
+                "object": {
+                    "customer": "cus_REbNQXKKFCRF2c",
+                }
+            },
+        }
+
+        CustomerDetail.objects.create(
+            user=user, stripe_customer_id=payload["data"]["object"]["customer"]
+        )
+
+        Subscription.objects.create(user=user, status="active")
+
+        sig_header = generate_stripe_signature(payload, STRIPE_WEBHOOK_SECRET)
+        mock_construct_event.return_value = payload
+
+        response = client.post(
+            reverse_lazy("v1:payment:stripe-webhook"),
```

Comment:
keep the same way of working as the other tests of the code base.
Text the endpoint URL and then simply use self.url in the other tests, group the tests per endpoints

Reply:
I've modified the tests to keep the same approach as the other tests in the code base